### PR TITLE
Add test for scrolling in Quirks and nonQuirks mode.

### PR DESCRIPTION
**Pull request by fred-wang**
@zcorpan I'm just opening a PR to discuss it and get some preliminary feedback but I believe the tests can be extended / improved. I'll go back to this later.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/5637)

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5637/)

File: cssom-view/scrolling-quirks-vs-nonquirks.html [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  body.style.margin = 0;
+  body.innerHTML = "<div id='content' style='width: 700px; height: 900px; background: linear-gradient(135deg, red, blue);'></div>";
+}
+
+async_test(function() {
+  var quirksFrame = document.getElementById("quirksframe");
+  quirksFrame.onload = this.step_func_done(function() {
+    var doc = quirksFrame.contentDocument;
+    assert_equals(doc.compatMode, "BackCompat", "Should be in quirks mode.");
+    assert_not_equals(doc.body, null, "Should have a body element");
```
Comment (zcorpan) [inline cssom-view/scrolling-quirks-vs-nonquirks.html]: Can remove this... if it's null, it will throw in `setBodyContent` anyway.

**Review by zcorpan — COMMENTED** [review]
Great with tests for this! Indeed there are several aspects to test here... could maybe use https://github.com/mathiasbynens/document.scrollingElement/tree/master/tests for inspiration

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/224385319)

# Lint #

Passed

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/224385321)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 3431987240288994ad1a898b5fc000f935a30037
Using browser at version BuildID 20170420100256; SourceStamp 27311156637f9b5d4504373967e01c4241902ae7
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5637/cssom-view/scrolling-quirks-vs-nonquirks.html">/cssom-view/scrolling-quirks-vs-nonquirks.html</a></summary>


|            Subtest            | Results | Messages |
|-------------------------------|---------|----------|
|                               | OK      |          |
| `scrolling in quirks mode`    | PASS    |          |
| `scrolling in no-quirks mode` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/224385328)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 3431987240288994ad1a898b5fc000f935a30037
Using browser at version 59.0.3071.15 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5637/cssom-view/scrolling-quirks-vs-nonquirks.html">/cssom-view/scrolling-quirks-vs-nonquirks.html</a></summary>


|            Subtest            | Results |                    Messages                    |
|-------------------------------|---------|------------------------------------------------|
|                               | OK      |                                                |
| `scrolling in quirks mode`    | FAIL    | `doc.documentElement.scroll is not a function` |
| `scrolling in no-quirks mode` | FAIL    | `doc.documentElement.scroll is not a function` |

</details>
</details>

**Comment by fred-wang** [comment]
@zcorpan OK, I've cleaned things up a bit in a second commit. However, I believe I'm finally going to stick with the current set of tests for this PR (as these are the problematic ones in Chromium & WebKit releases) and will consider other cases later.

The tests pass for me on Gecko as well as on Chromium with the ScrollTopLeftInterop flag turned on (cc'ing @RByers who worked on the latter).

> could maybe use https://github.com/mathiasbynens/document.scrollingElement/tree/master/tests for inspiration

These are already handled by https://w3c-test.org/cssom-view/scrollingElement.html, right?

**Comment by RByers** [comment]
This is great, thank you!  It covers the most important use cases.  There are additional edge cases that could be valuable to cover (eg. the weird quirks behavior with 'html, body { overflow: scroll; }' where IIRC the only way to get/set the real document scroll position is with window.scroll*).  See, for example, our [tests for this in blink](https://cs.chromium.org/chromium/src/third_party/WebKit/LayoutTests/fast/dom/Element/scrollTop-scrollLeft-body.html?dr). But it's obviously up to you if you think it's worthwhile to add any of those edge cases.

**Comment by RByers** [comment]
Oh, I missed that you said you'll consider other cases in a separate PR.  SGTM.

**Review by RByers — APPROVED** [review]

**Comment by fred-wang** [comment]
@RByers Thanks. I guess it will be worth trying to convert Blink tests into WPT tests at some point. There are other things I was thinking like:
- testing clientWidth, clientHeight scrollWidth, scrollHeight
- testing the case when the "HTML body element in quirks mode is potentially scrollable" (which is maybe the one you mentioned)
- maybe writing reftests to test setter without using the getter (and conversely)

However, I'll probably only go back to this once I experiment with importing the code from Chromium to WebKit first so that I get a better idea on this.

**Pull request merged**